apps/prairielearn/src/pages/studentInstanceQuestion/studentInstanceQuestion.html.ts

```diff
@@ -0,0 +1,171 @@
+        ${resLocals.assessment.type === 'Exam'
+          ? html`
+              ${compiledScriptTag('examTimeLimitCountdown.ts')}
+              ${EncodedData(
+                {
+                  serverRemainingMS: resLocals.assessment_instance_remaining_ms,
+                  serverTimeLimitMS: resLocals.assessment_instance_time_limit_ms,
+                  serverUpdateURL:
+                    resLocals.urlPrefix +
+                    '/assessment_instance/' +
+                    resLocals.assessment_instance.id +
+                    '/time_remaining',
+                  canTriggerFinish: resLocals.authz_result.authorized_edit,
+                  csrfToken: resLocals.__csrf_token,
+                },
+                'time-limit-data',
+              )}
+            `
+          : ''}
+        <script defer src="${nodeModulesAssetPath('mathjax/es5/startup.js')}"></script>
+        <script>
+          document.urlPrefix = '${resLocals.urlPrefix}';
+        </script>
+        ${resLocals.no_variant_exists
+          ? ''
+          : html`
+              ${resLocals.question.type !== 'Freeform'
+                ? html`
+                    <script src="${nodeModulesAssetPath('lodash/lodash.min.js')}"></script>
+                    <script src="${assetPath('javascripts/require.js')}"></script>
+                    <script src="${assetPath('localscripts/question.js')}"></script>
+                    <script src="${assetPath(
+                        `localscripts/question${resLocals.effectiveQuestionType}.js`,
+                      )}"></script>
+                  `
+                : ''}
+              ${unsafeHtml(resLocals.extraHeadersHtml)}
+            `}
+      </head>
+      <body>
+        <script>
+          // make the file inputs display the file name
+          $(document).on('change', '.custom-file-input', function () {
+              let fileName = $(this).val().replace(/\\/g, '/').replace(/.*//, '');
+              $(this).parent('.custom-file').find('.custom-file-label').text(fileName);
+          });
+        </script>
+
+        ${renderEjs(import.meta.url, "<%- include('../partials/navbar'); %>", {
+          ...resLocals,
+          navPage: '',
+        })}
+        <main id="content" class="container">
+          <div class="row">
+            <div class="col-lg-9 col-sm-12">
+              ${resLocals.no_variant_exists
+                ? html`
+                    <div class="card mb-4">
+                      <div class="card-header bg-primary text-white">
+                        ${renderEjs(
+                          import.meta.url,
+                          "<%- include('../partials/questionTitle') %>",
+                          {
+                            ...resLocals,
+                            question_context:
+                              resLocals.assessment.type === 'Exam'
+                                ? 'student_exam'
+                                : 'student_homework',
```

Comment:
Nit: this is computed in multiple places, maybe create a variable to avoid repetition?

Reply:
Done in c72e14a.